redisson/src/main/java/org/redisson/misc/LogHelper.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2016 Nikita Koksharov
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.redisson.misc;
+
+import java.lang.reflect.Array;
+import java.util.Collection;
+import java.util.Objects;
+
+/**
+ * @author Philipp Marx
+ */
+public class LogHelper {
+
+    private static final int MAX_COLLECTION_LOG_SIZE = Integer.valueOf(System.getProperty("redisson.maxCollectionLogSize", "10"));
+    private static final int MAX_STRING_LOG_SIZE = Integer.valueOf(System.getProperty("redisson.maxStringLogSize", "100"));
+
+    private LogHelper() {
+    }
+
+    public static <T> String toString(T object) {
+        if (object == null) {
+            return "null";
+        } else if (object instanceof String) {
+            return toStringString((String) object);
+        } else if (object.getClass().isArray()) {
+            return toArrayString(object);
+        } else if (object instanceof Collection) {
+            return toCollectionString((Collection<?>) object);
+        } else {
+            return Objects.toString(object);
+        }
+    }
+
+    private static String toStringString(String string) {
+        if (string.length() > MAX_STRING_LOG_SIZE) {
+            return new StringBuilder(string.substring(0, MAX_STRING_LOG_SIZE)).append("...").toString();
+        } else {
+            return string;
+        }
+    }
+
+    private static String toCollectionString(Collection<?> collection) {
+        return toArrayString(collection.toArray(new Object[collection.size()]));
```

Comment:
I had given this a little though when implementing this. I don't think this is too big of an issue though since the only memory which needs to be allocated is the size of the array (+ the number of pointers within the array). The content is already in memory and is referenced by the collection anyway (as collections don't contain primitives, the array won't either). The array is eligible for GC right afterwards anyway so would be GC'ed in case of OOMs. Of course the created array could already be capped in its size with MAX_COLLECTION_LOG_SIZE + 1 (to get the ellipses printed).

Another issue though is, that Maps should be considered as well.